### PR TITLE
Cewood/misc fixes

### PR DESCRIPTION
**Pull request by cewood**

### CODE REVIEW

**Comment by cewood** [comment]
Very early days, just thought I'd start the conversation now.

File: main.go [hunk 1]
```diff
@@ -9,26 +10,38 @@ import (
...
-	if err != nil {
-		return false
-	}
+		defer response.Body.Close()
 
-	if response.StatusCode < 200 || response.StatusCode >= 300 {
-		return false
+		if response.StatusCode < 200 || response.StatusCode >= 300 {
+			err := errors.New("Returned status code was not acceptable")
+			return err
```
Comment (nchern) [inline main.go]: return errors.New("Returned status code was not acceptable")

File: main.go [hunk 2]
```diff
@@ -9,26 +10,38 @@ import (
...
-	// Make request
-	response, err := http.Get("https://www.google.com")
+		if err != nil {
+			return err
+		}
 
-	if err != nil {
-		return false
-	}
+		defer response.Body.Close()
```
Comment (nchern) [inline main.go:41]: we need to read a response body even if we are not going to use it. Like this: `io.Copy(ioutil.Discard, response.Body)`
http://stackoverflow.com/questions/17948827/reusing-http-connections-in-golang

File: main.go [hunk 3]
```diff
@@ -9,26 +10,38 @@ import (
...
+		if response.StatusCode < 200 || response.StatusCode >= 300 {
+			err := errors.New("Returned status code was not acceptable")
+			return err
+		}
+		return nil // or an error
 	}
 
-	return true
+	err := backoff.Retry(operation, backoff.NewExponentialBackOff())
+	if err != nil {
```
Comment (nchern) [inline main.go]: `return err == nil`

File: retry/retry.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	// custExpBackOff := &backoff.ExponentialBackOff{
+	// 	InitialInterval:     backoff.DefaultInitialInterval,
+	// 	RandomizationFactor: 0,
+	// 	Multiplier:          0,
+	// 	MaxInterval:         500 * time.Millisecond,
+	// 	MaxElapsedTime:      10 * time.Second,
+	// }
+
+	customBackoff := backoff.NewExponentialBackOff()
+	customBackoff.MaxElapsedTime = 10 * time.Second
```
Comment (nchern) [inline retry/retry.go]: to constants

File: services/service.go [hunk 5]
```diff
@@ -59,60 +61,77 @@ func (s *Service) RunAll() error {
...
 
-	if response.StatusCode != http.StatusOK {
-		return fmt.Errorf("Service HTTP status code is %d, %d expected", response.StatusCode, http.StatusOK)
-	}
+		if err != nil {
+			return err
+		}
 
-	return nil
+		defer response.Body.Close()
```
Comment (nchern) [inline services/service.go:73]: same here, we need to read the body

File: main.go (hunk 2 above)
Comment (cewood) [reply main.go:41]: DONE

File: main.go (hunk 1 above)
Comment (cewood) [reply main.go]: DONE

File: main.go (hunk 3 above)
Comment (cewood) [reply main.go]: DONE

File: retry/retry.go (hunk 4 above)
Comment (cewood) [reply retry/retry.go]: DONE

File: services/service.go (hunk 5 above)
Comment (cewood) [reply services/service.go:73]: DONE

**Comment by nchern** [comment]
lgtm

**Pull request merged**